utils/crates/common/src/contracts.rs

```diff
@@ -0,0 +1,215 @@
+use alloy::{
+    primitives::{b256, keccak256, FixedBytes},
+    sol,
+};
+use IConstantProductHelper::OnchainOrder;
+
+// Codegen from ABI file to interact with the contract.
+sol!(
+    #[allow(missing_docs)]
+    #[allow(clippy::too_many_arguments)]
+    #[derive(Debug)]
+    #[sol(rpc)]
+    ComposableCoW,
+    "../../../out/ComposableCoW.sol/ComposableCoW.json"
+);
+
+sol!(
+    #[allow(missing_docs)]
+    #[allow(clippy::too_many_arguments)]
+    #[derive(Debug)]
+    #[sol(rpc)]
+    GPv2Settlement,
+    "../../../out/GPv2Settlement.sol/GPv2Settlement.json"
+);
+
+sol!(
+    #[allow(missing_docs)]
+    #[derive(Debug)]
+    #[sol(rpc)]
+    interface IERC20 {
+        function balanceOf(address account) external view returns (uint256);
+    }
```

Comment:
Unrelated to this PR: I remember there was some discussion about removing `ConditionalCoW` from this repo and the artifacts being needed. Maybe we could use a trick like this with the few functions we need?

Reply:
I think the main reason for keeping it around is that some of the test suite from `ComposableCoW` is used in the E2E tests.